src/MSBuild/LiveLogger/Terminal.cs

```diff
@@ -29,18 +30,49 @@ internal sealed class Terminal : ITerminal
     /// </summary>
     private bool _isBuffering = false;
 
+    internal TextWriter Output { private get; set; } = Console.Out;
+
     /// <inheritdoc/>
-    public int Height => Console.BufferHeight;
+    public int Height
+    {
+        get
+        {
+            if (Console.IsOutputRedirected)
+            {
+                return int.MaxValue;
```

Comment:
Should we default to "infinite", like I've done here, or to "a reasonable size like 80×20"?

Reply:
For redirected output it makes sense to use infinite. For tests I think we may want to use reasonable (as well as unreasonable!) values. I would probably add `int?` parameters to the new constructor and default to infinite if null.

Reply:
I agree--we should manipulate that to test the selective-truncation stuff. I'll leave that for later.